src/test/optimizer/strategy/subquery_to_join_rule_test.cpp

```diff
@@ -0,0 +1,394 @@
+#include "gtest/gtest.h"
+
+#include "strategy_base_test.hpp"
+#include "testing_assert.hpp"
+
+#include "expression/expression_functional.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/limit_node.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "optimizer/strategy/subquery_to_join_rule.hpp"
+#include "storage/storage_manager.hpp"
+#include "utils/load_table.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+class SubqueryToJoinRuleTest : public StrategyBaseTest {
+ public:
+  void SetUp() override {
+    node_a = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}}, "a");
+    a_a = {node_a, ColumnID{0}};
```

Comment:
you could also do `node_a->get_column("a")`, which imo reads a bit nicer and is safer